examples/complete/s3-origins.tf

```
@@ -0,0 +1,54 @@
+locals {
+  additional_s3_origins_enabled = local.enabled && var.additional_s3_origins_enabled
+  default_s3_origin_configuration = {
+    domain_name = null
+    origin_id   = null
+    origin_path = null
+    s3_origin_config = {
+      origin_access_identity = ""
+    }
+  }
+  additional_s3_origins = local.additional_s3_origins_enabled ? [
+    merge(local.default_s3_origin_configuration, {
+      domain_name = module.additional_s3_origin.bucket_regional_domain_name
+      origin_id   = module.additional_s3_origin.bucket_id
+    }),
+    merge(local.default_s3_origin_configuration, {
+      domain_name = module.additional_s3_failover_origin.bucket_regional_domain_name
+      origin_id   = module.additional_s3_failover_origin.bucket_id
+    })
+  ] : []
+  additional_s3_origin_groups = local.additional_s3_origins_enabled ? [{
+    primary_origin_id  = local.additional_s3_origins[0].origin_id
+    failover_origin_id = local.additional_s3_origins[1].origin_id
+    failover_criteria  = var.origin_group_failover_criteria_status_codes
+  }] : []
```

nit: It would be more readable to break this out into separate primary and secondary origins but feel free to ignore if it further complicates things.

```suggestion
  additional_s3_origins_primary = local.additional_s3_origins_enabled ? merge(
    local.default_s3_origin_configuration, {
      domain_name = module.additional_s3_origin.bucket_regional_domain_name
      origin_id   = module.additional_s3_origin.bucket_id
    },
  ) : {}
  additional_s3_origins_secondary = local.additional_s3_origins_enabled ? merge(
    local.default_s3_origin_configuration, {
      domain_name = module.additional_s3_failover_origin.bucket_regional_domain_name
      origin_id   = module.additional_s3_failover_origin.bucket_id
    }
  ) : {}
  additional_s3_origin_groups = local.additional_s3_origins_enabled ? [{
    primary_origin_id  = local.additional_s3_origins_primary.origin_id
    failover_origin_id = local.additional_s3_origins_secondary.origin_id
    failover_criteria  = var.origin_group_failover_criteria_status_codes
  }] : []
```